sdk/src/resource/resource.cc

```diff
@@ -25,15 +27,19 @@ Resource Resource::Merge(const Resource &other) noexcept
 Resource Resource::Create(const ResourceAttributes &attributes)
 {
   static auto otel_resource = OTELResourceDetector().Detect();
-  auto default_resource     = Resource::GetDefault();
+  auto resource = Resource::GetDefault().Merge(otel_resource).Merge(Resource(attributes));
```

Comment:
Is `attributes.size()` still needs to be checked before creating `Resource` based on it?

Reply:
No, it should be fine to let merge empty resource and remove this check.

Reply:
Added test to validate resource creation with empty attributes.